PayloadPkg/FirmwareUpdate/FirmwareUpdate.c

```diff
@@ -95,79 +95,98 @@ VerifyFwVersion (
+    } else if (FwPolicy.Fields.UpdatePartitionA == 0x1) {
+      Status = GetComponentInfoByPartition(FLASH_MAP_SIG_STAGE1A, TRUE, &CompBase, &CompSize);
+    }
 
-  //
-  // Get base address of Stage 1A in capsule Image
-  //
-  if (FwPolicy.Fields.UpdatePartitionB == 0x1) {
-    Status = PlatformGetStage1AOffset(ImageHdr, FALSE, &CompBase, &CompSize);
-  } else if (FwPolicy.Fields.UpdatePartitionA == 0x1) {
-    Status = PlatformGetStage1AOffset(ImageHdr, TRUE, &CompBase, &CompSize);
-  }
-  if (EFI_ERROR (Status)) {
-    DEBUG((DEBUG_ERROR, "PlatformGetStage1AOffset: %r\n", Status));
-    return Status;
-  }
+    if (EFI_ERROR(Status)) {
+      DEBUG((DEBUG_ERROR, "GetComponentInfoByPartition: %r\n", Status));
+      return Status;
+    }
 
-  Status = GetVersionfromFv (&CompBase, &CapsuleBlVersion);
-  if (EFI_ERROR (Status)) {
-    DEBUG((DEBUG_ERROR, "GetVersionfromFv: %r\n", Status));
-    return Status;
-  }
+    Status = GetVersionfromFv (&CompBase, &CurrentBlVersion);
+    if (EFI_ERROR (Status)) {
+      DEBUG((DEBUG_ERROR, "GetVersionfromFv: %r\n", Status));
+      return Status;
+    }
 
-  //
-  // Update last update version to the version we are about to update
-  //
-  Status = UpdateStatus(ImageHdr->UpdateHardwareInstance, \
-                        (CapsuleBlVersion->ImageVersion.ProjMajorVersion << 8) | CapsuleBlVersion->ImageVersion.ProjMinorVersion, \
-                        0xFFFFFFFF);
-  if (EFI_ERROR (Status)) {
-    DEBUG((DEBUG_ERROR, "Updating status to reserved region failed: %r\n", Status));
-    return Status;
-  }
+    //
+    // Get base address of Stage 1A in capsule Image
+    //
+    if (FwPolicy.Fields.UpdatePartitionB == 0x1) {
+      Status = PlatformGetStage1AOffset(ImageHdr, FALSE, &CompBase, &CompSize);
+    } else if (FwPolicy.Fields.UpdatePartitionA == 0x1) {
+      Status = PlatformGetStage1AOffset(ImageHdr, TRUE, &CompBase, &CompSize);
+    }
+    if (EFI_ERROR (Status)) {
+      DEBUG((DEBUG_ERROR, "PlatformGetStage1AOffset: %r\n", Status));
+      return Status;
+    }
 
-  if (CapsuleBlVersion->ImageVersion.SecureVerNum >= CurrentBlVersion->ImageVersion.SecureVerNum) {
-    return EFI_SUCCESS;
-  }
+    Status = GetVersionfromFv (&CompBase, &CapsuleBlVersion);
+    if (EFI_ERROR (Status)) {
+      DEBUG((DEBUG_ERROR, "GetVersionfromFv: %r\n", Status));
+      return Status;
+    }
+
+    //
+    // Update last update version to the version we are about to update
+    //
+    Status = UpdateStatus(ImageHdr->UpdateHardwareInstance, \
+                          (CapsuleBlVersion->ImageVersion.ProjMajorVersion << 8) | CapsuleBlVersion->ImageVersion.ProjMinorVersion, \
+                          0xFFFFFFFF);
+    if (EFI_ERROR (Status)) {
+      DEBUG((DEBUG_ERROR, "Updating status to reserved region failed: %r\n", Status));
+      return Status;
+    }
+
+    if (CapsuleBlVersion->ImageVersion.SecureVerNum >= CurrentBlVersion->ImageVersion.SecureVerNum) {

```

Comment:
How is this SVN different from the SVN stored in BtG BPM ?

Reply:
@sagraw2 This **SecureVerNum** is part of Stage1A flash region which gets included at build image generation as part of ImageVer details. This is not fused and it would be overwritten with flash region update.
**SecureVerNum** happens at capsule firmware update where as Btg BPM SVN will be fused and it would be an boot time check. Ideally **SecureVerNum** and BtgSVN should be in sync.